### PR TITLE
sysvars: add `tidb_shard_row_id_bits` and `tidb_pre_split_regions`

### PR DESCRIPTION
**Pull request by jackysp**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Frank945946 — APPROVED** [review]

File: system-variables.md [hunk 1]
```diff
@@ -4305,6 +4305,16 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
 
+### `tidb_pre_split_regions` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`0`
+- 范围：`[0, 15]`
+- 该变量用于设置新建表的默认行分裂分片数。当设置了该变量为非 0 值后，执行 CRATE TABLE 语句时，TiDB 会将为允许使用 `pre_split_regions` 的表 (如，NONCLUSTERED 表) 自动设定该属性。详细说明见 [pre_split_regions](/sql-statements/sql-statement-split-region.md)。该变量通常与 `tidb_shard_row_id_bits` 配合使用，用于为新建表进行分片以及 region 预分裂。
```
Comment (hfxsd) [inline system-variables.md:4316]:
```suggestion
- 该变量用于设置新建表的默认行分裂分片数。当设置了该变量为非 0 值后，执行 `CRATE TABLE` 语句时，TiDB 会将为允许使用 `pre_split_regions` 的表（如，NONCLUSTERED 表）自动设定该属性。详细说明见 [pre_split_regions](/sql-statements/sql-statement-split-region.md)。该变量通常与 [`tidb_shard_row_id_bits`](/system-variables.md#tidb_shard_row_id_bits-从-v840-版本开始引入) 配合使用，用于为新建表进行分片以及 Region 预分裂。
```

File: system-variables.md [hunk 2]
```diff
@@ -4602,6 +4612,16 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
 
+### `tidb_shard_row_id_bits` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`0`
+- 范围：`[0, 15]`
+- 该变量用于设置新建表的默认行 ID 的分片位数。当设置了该变量为非 0 值后，执行 CRATE TABLE 语句时，TiDB 会将为允许使用 `shard_row_id_bits` 的表 (如，NONCLUSTERED 表) 自动设定该属性。详细说明见 [SHARD_ROW_ID_BITS](/shard-row-id-bits.md)。
```
Comment (hfxsd) [inline system-variables.md:4623]:
```suggestion
- 该变量用于设置新建表的默认行 ID 的分片位数。当设置了该变量为非 0 值后，执行 `CRATE TABLE` 语句时，TiDB 会将为允许使用 `shard_row_id_bits` 的表（如，NONCLUSTERED 表）自动设定该属性。详细说明见 [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md)。
```

File: system-variables.md [hunk 3]
```diff
@@ -4602,6 +4612,16 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
 
+### `tidb_shard_row_id_bits` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`0`
+- 范围：`[0, 15]`
+- 该变量用于设置新建表的默认行 ID 的分片位数。当设置了该变量为非 0 值后，执行 `CRATE TABLE` 语句时，TiDB 会将为允许使用 `SHARD_ROW_ID_BITS` 的表（如，NONCLUSTERED 表）自动设定该属性。详细说明见 [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md)。
```
Comment (hfxsd) [inline system-variables.md:4623]:
```suggestion
- 该变量用于设置新建表的默认行 ID 的分片位数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会将为允许使用 `SHARD_ROW_ID_BITS` 的表（如，NONCLUSTERED 表）自动设定该属性。详细说明见 [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md)。
```

File: system-variables.md [hunk 4]
```diff
@@ -4305,6 +4305,16 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
 
+### `tidb_pre_split_regions` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`0`
+- 范围：`[0, 15]`
+- 该变量用于设置新建表的默认行分裂分片数。当设置了该变量为非 0 值后，执行 `CRATE TABLE` 语句时，TiDB 会将为允许使用 `PRE_SPLIT_REGIONS` 的表（如，NONCLUSTERED 表）自动设定该属性。详细说明见 [`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions)。该变量通常与 [`tidb_shard_row_id_bits`](/system-variables.md#tidb_shard_row_id_bits-从-v840-版本开始引入) 配合使用，用于为新建表进行分片以及 Region 预分裂。
```
Comment (hfxsd) [inline system-variables.md:4316]:
```suggestion
- 该变量用于设置新建表的默认行分裂分片数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会将为允许使用 `PRE_SPLIT_REGIONS` 的表（如，NONCLUSTERED 表）自动设定该属性。详细说明见 [`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions)。该变量通常与 [`tidb_shard_row_id_bits`](/system-variables.md#tidb_shard_row_id_bits-从-v840-版本开始引入) 配合使用，用于为新建表进行分片以及 Region 预分裂。
```

File: system-variables.md [hunk 5]
```diff
@@ -4602,6 +4612,16 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
 
+### `tidb_shard_row_id_bits` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`0`
+- 范围：`[0, 15]`
+- 该变量用于设置新建表的默认行 ID 的分片位数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会将为允许使用 `SHARD_ROW_ID_BITS` 的表（如，NONCLUSTERED 表）自动设定该属性。详细说明见 [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md)。
```
Comment (hfxsd) [inline system-variables.md:4623]:
```suggestion
- 该变量用于设置新建表的默认行 ID 的分片位数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会将为允许使用 `SHARD_ROW_ID_BITS` 的表（如 NONCLUSTERED 表）自动设定该属性。详细说明见 [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md)。
```

File: system-variables.md [hunk 6]
```diff
@@ -4305,6 +4305,16 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
 
+### `tidb_pre_split_regions` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`0`
+- 范围：`[0, 15]`
+- 该变量用于设置新建表的默认行分裂分片数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会将为允许使用 `PRE_SPLIT_REGIONS` 的表（如，NONCLUSTERED 表）自动设定该属性。详细说明见 [`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions)。该变量通常与 [`tidb_shard_row_id_bits`](/system-variables.md#tidb_shard_row_id_bits-从-v840-版本开始引入) 配合使用，用于为新建表进行分片以及 Region 预分裂。
```
Comment (hfxsd) [inline system-variables.md:4316]:
```suggestion
- 该变量用于设置新建表的默认行分裂分片数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会为允许使用 `PRE_SPLIT_REGIONS` 的表（如，NONCLUSTERED 表）自动设定该属性。详细说明见 [`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions)。该变量通常与 [`tidb_shard_row_id_bits`](/system-variables.md#tidb_shard_row_id_bits-从-v840-版本开始引入) 配合使用，用于为新建表进行分片以及 Region 预分裂。
```

File: system-variables.md [hunk 7]
```diff
@@ -4602,6 +4612,16 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
 
+### `tidb_shard_row_id_bits` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`0`
+- 范围：`[0, 15]`
+- 该变量用于设置新建表的默认行 ID 的分片位数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会将为允许使用 `SHARD_ROW_ID_BITS` 的表（如 NONCLUSTERED 表）自动设定该属性。详细说明见 [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md)。
```
Comment (hfxsd) [inline system-variables.md:4623]:
```suggestion
- 该变量用于设置新建表的默认行 ID 的分片位数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会为允许使用 `SHARD_ROW_ID_BITS` 的表（如 NONCLUSTERED 表）自动设定该属性。详细说明见 [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md)。
```

**Review by hfxsd — APPROVED** [review]

File: system-variables.md [hunk 8]
```diff
@@ -4305,6 +4305,16 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
 
+### `tidb_pre_split_regions` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`0`
+- 范围：`[0, 15]`
+- 该变量用于设置新建表的默认行分裂分片数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会为允许使用 `PRE_SPLIT_REGIONS` 的表（如，NONCLUSTERED 表）自动设定该属性。详细说明见 [`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions)。该变量通常与 [`tidb_shard_row_id_bits`](/system-variables.md#tidb_shard_row_id_bits-从-v840-版本开始引入) 配合使用，用于为新建表进行分片以及 Region 预分裂。
```
Comment (lilin90) [inline system-variables.md:4316]:
```suggestion
- 该变量用于设置新建表的默认行分裂分片数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会为允许使用 `PRE_SPLIT_REGIONS` 的表（例如 `NONCLUSTERED` 表）自动设定该属性。详见 [`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions)。该变量通常与 [`tidb_shard_row_id_bits`](/system-variables.md#tidb_shard_row_id_bits-从-v840-版本开始引入) 配合使用，用于为新建表进行分片以及 Region 预分裂。
```

File: system-variables.md [hunk 9]
```diff
@@ -4602,6 +4612,16 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
 
+### `tidb_shard_row_id_bits` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`0`
+- 范围：`[0, 15]`
+- 该变量用于设置新建表的默认行 ID 的分片位数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会为允许使用 `SHARD_ROW_ID_BITS` 的表（如 NONCLUSTERED 表）自动设定该属性。详细说明见 [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md)。
```
Comment (lilin90) [inline system-variables.md:4623]:
```suggestion
- 该变量用于设置新建表的默认行 ID 的分片位数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会为允许使用 `SHARD_ROW_ID_BITS` 的表（例如 `NONCLUSTERED` 表）自动设定该属性。详见 [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md)。
```

File: system-variables.md [hunk 10]
```diff
@@ -4305,6 +4305,16 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
 
+### `tidb_pre_split_regions` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`0`
+- 范围：`[0, 15]`
+- 该变量用于设置新建表的默认行分裂分片数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会为允许使用 `PRE_SPLIT_REGIONS` 的表（例如 `NONCLUSTERED` 表）自动设定该属性。详见 [`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions)。该变量通常与 [`tidb_shard_row_id_bits`](/system-variables.md#tidb_shard_row_id_bits-从-v840-版本开始引入) 配合使用，用于为新建表进行分片以及 Region 预分裂。
```
Comment (lilin90) [inline system-variables.md:4316]:
```suggestion
- 该变量用于设置新建表默认的行分裂分片数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会为允许使用 `PRE_SPLIT_REGIONS` 的表（例如 `NONCLUSTERED` 表）自动设定该属性。详见 [`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions)。该变量通常与 [`tidb_shard_row_id_bits`](/system-variables.md#tidb_shard_row_id_bits-从-v840-版本开始引入) 配合使用，用于为新建表进行分片以及 Region 预分裂。
```

File: system-variables.md [hunk 11]
```diff
@@ -4602,6 +4612,16 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
 
+### `tidb_shard_row_id_bits` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`0`
+- 范围：`[0, 15]`
+- 该变量用于设置新建表的默认行 ID 的分片位数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会为允许使用 `SHARD_ROW_ID_BITS` 的表（例如 `NONCLUSTERED` 表）自动设定该属性。详见 [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md)。
```
Comment (lilin90) [inline system-variables.md:4623]:
```suggestion
- 该变量用于设置新建表默认的行 ID 的分片数。当设置了该变量为非 0 值后，执行 `CREATE TABLE` 语句时，TiDB 会为允许使用 `SHARD_ROW_ID_BITS` 的表（例如 `NONCLUSTERED` 表）自动设定该属性。详见 [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md)。
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**